OpenRA.Mods.Common/ActorExts.cs

```diff
@@ -82,5 +82,24 @@ public static CPos ClosestCell(this Actor self, IEnumerable<CPos> cells)
 		{
 			return cells.MinByOrDefault(c => (self.Location - c).LengthSquared);
 		}
+
+		public static DamageState GetDamageState(this Actor self)
+		{
+			if (self.WillDispose)
+				return DamageState.Dead;
+
+			var health = self.TraitOrDefault<Health>();
```

Comment:
We could avoid these queries by making `Actor.death` public and doing `self.death as IHealth` - but this looks awkward so we should try and come up with a different name.